File: client-app/src/app/models/activity.ts
import { Profile } from "./profile";

export interface Activity {
    id: string;
    title: string;
    description: string;
    category: string;
    date: Date | null;
    city: string;
    venue: string;
    options: string[];
    winningOption: string;
    amountIfWon: number;
    amountIfLose: number;
    isMandatoryActivity: boolean;
    hostUsername?: string;
    isCancelled?: boolean;
    isGoing?: boolean;
    isHost?: boolean
    attendees: Profile[]
    host?: Profile;    
}

export class ActivityFormValues
  {
    id?: string = undefined;
    title: string = '';
    category: string = '';
    description: string = '';
    date: Date | null = null;
    city: string = '';
    venue: string = '';
    options: string[] = [];
    winningOption: string = '';
    amountIfWon: number = 0 ;
    amountIfLose: number = 0;
    isMandatoryActivity: boolean = false;

	  constructor(activity?: ActivityFormValues) {
      if (activity) {
        this.id = activity.id;
        this.title = activity.title;
        this.category = activity.category;
        this.description = activity.description;
        this.date = activity.date;
        this.venue = activity.venue;
        this.city = activity.city;
        this.options = activity.options;
        this.winningOption = activity.winningOption;
        this.amountIfWon = activity.amountIfWon;
        this.amountIfLose = activity.amountIfLose;
        this.isMandatoryActivity = activity.isMandatoryActivity;
      }
    }

  }

  export class Activity implements Activity {
    
  constructor(init?: ActivityFormValues) {
    if (init) {
      Object.assign(this, init);
    }
  }
  }

File: Domain/Activity.cs
namespace Domain;

public class Activity
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public DateTime Date { get; set; }
    public string Description { get; set; }
    public string Category { get; set; }
    public string City { get; set; }
    public string Venue { get; set; }
    public bool IsCancelled { get; set; }

    public ICollection<ActivityAttendee> Attendees { get; set; } = new List<ActivityAttendee>();
    public ICollection<Comment> Comments { get; set; } = new List<Comment>();

    public string[] Options { get; set; }

    public string WinningOption { get; set; }

    public int AmountIfWon { get; set; }

    public int AmountIfLose { get; set; }

    public bool IsMandatoryActivity { get; set; }
}